core/src/main/scala/kafka/controller/PartitionStateMachine.scala

```diff
@@ -470,7 +469,7 @@ class PartitionStateMachine(controller: KafkaController) extends Logging {
           nonExistentTopics.foreach(topic => zkUtils.deletePathRecursive(getDeleteTopicPath(topic)))
         }
         topicsToBeDeleted --= nonExistentTopics
-        if(topicsToBeDeleted.size > 0) {
+        if(controller.config.deleteTopicEnable && topicsToBeDeleted.size > 0) {
```

Comment:
If we do the check here, the previous logs may still be printed. Maybe it is clearer to have a separate `DisabledDeleteTopicListener` class that simply log a warning message and delete the delete topic path.